src/onehot.jl

```diff
@@ -126,8 +126,18 @@ julia> onecold([ 1  0  0  1  0  1  0  1  0  0  1
 "abeacadabea"
 ```
 """
-onecold(y::AbstractVector, labels = 1:length(y)) = labels[argmax(y)]
+function onecold(y::AbstractVector, labels = 1:length(y))
+  nl = length(labels)
+  ny = length(y)
+  nl == ny || throw(DimensionMismatch("onecold got $nl lables for a vector of length $ny, these must agree"))
```

Comment:
```suggestion
  nl == ny || throw(DimensionMismatch("onecold got $nl labels for a vector of length $ny, these must agree"))
```
Would it make sense to match the format of other `DimensionMismatch` errors in Base? e.g.:
```julia-repl
julia> rand(3, 1) * rand(2, 1)
ERROR: DimensionMismatch: A has dimensions (3,1) but B has dimensions (2,1)
```

Reply:
IDK how much we gain from consistency. Maybe better not to remind people precisely of the matmul error? Another common one is:
```
julia> rand(3, 1) .* rand(2, 1)
ERROR: DimensionMismatch: arrays could not be broadcast to a common size; got a dimension with lengths 3 and 2
```
This one says it's about broadcasting, which seems helpful, so I put "onecold" in the words

Reply:
That broadcasting error is a personal pet peeve for how little information it provides...My thought wasn't to replicate the matmul message, but to find a way around saying ", these must agree".